src/Control/Distributed/Process/Internal/Primitives.hs

```diff
@@ -400,8 +400,10 @@ newtype Match b = Match { unMatch :: MatchOn Message (Process b) }
 receiveWait :: [Match b] -> Process b
 receiveWait ms = do
   queue <- processQueue <$> ask
-  Just proc <- liftIO $ dequeue queue Blocking (map unMatch ms)
-  proc
+  mProc <- liftIO $ dequeue queue Blocking (map unMatch ms)
+  case mProc of
+    Just proc' -> proc'
+    Nothing    -> error "Well... That wasn't supposed to happen! o_O"
```

Comment:
@qnikst - clearly this isn't the right thing to do here! 🥇 

So what I can do is call the tracer here and debug log the fact that we saw an invariant in the cqueue code fail, and then call die with the appropriate message. That will kill the calling process in the correct way (with an exit signal). 

I'll cook that up and resubmit...